rero_ils/modules/documents/api.py

```diff
@@ -322,13 +322,13 @@ def replace_refs(self):
                     contribution['agent'] = agent
         subjects = self.get('subjects', [])
         for subject in subjects:
-            subject_ref = subject.get('$ref')
-            subject_type = subject.get('type')
-            if subject_ref and \
-                    subject_type in ['bf:Person', 'bf:Organisation']:
-                subject, _ = Contribution.get_record_by_ref(subject_ref)
-                if subject:
-                    subject.update({'type': subject_type})
+            ref = subject.get('$ref')
+            type = subject.get('type')
+            if ref and type in [DocumentSubjectType.PERSON,
+                                DocumentSubjectType.ORGANISATION]:
+                data, _ = Contribution.get_record_by_ref(ref)
+                del subject['$ref']
+                subject.update(data)
 
         return super().replace_refs()
```

Comment:
You're right, but I can't use the `replace_refs` because this function delete all keys from the current object to replace them with only the value of the reference. In my case, I need to keep some children keys (subdivisions, type, ...). The $ref resolution is done be the corresponding `ReferenceSubject` class.